src/DepositManager.mo

```diff
@@ -52,95 +51,58 @@ module {
   public class DepositManager(
     icrc84 : ICRC84Helper.Ledger,
     triggerOnNotifications : Bool,
-    changeCredit : ({ #pool; #user : Principal }, Int) -> (),
+    data : Data.Data,
+    feeManager : FeeManager.FeeManager,
     log : (Principal, LogEvent) -> (),
     trap : (text : Text) -> (),
   ) {
+    let { map; queue } = data;
+
     /// If `true` new notifications are paused.
     var paused : Bool = false;
 
-    /// Manages deposit balances for each user.
-    let depositRegistry = DepositRegistry.DepositRegistry(icrc84.fee(), trap);
-    let depositMap = depositRegistry.map;
-
     /// Total amount consolidated. Accumulated value.
     var totalConsolidated : Nat = 0;
 
+    /// Total amount of credited to users.
+    var totalCredited : Nat = 0;
+
     /// Total funds underway for consolidation.
     var underwayFunds : Nat = 0;
 
-    func surcharge() : Nat = depositRegistry.fee - icrc84.fee();
-
     public func state() : State = {
       paused = paused;
       fee = {
-        ledger = icrc84.fee();
-        deposit = depositRegistry.fee;
-        surcharge = surcharge();
+        ledger = feeManager.ledgerFee();
+        deposit = feeManager.fee();
+        surcharge = feeManager.surcharge();
       };
       flow = {
-        credited = Int.abs(credit.flow);
+        credited = totalCredited;
       };
       totalConsolidated = totalConsolidated;
       funds = {
-        deposited = depositMap.sum() + underwayFunds;
+        deposited = map.depositSum() + underwayFunds;
         underway = underwayFunds;
-        queued = depositMap.sum();
+        queued = map.depositSum();
       };
-      nDeposits = depositMap.size();
-      nLocks = depositMap.locks();
+      nDeposits = map.depositsCount();
+      nLocks = map.locks();
```

Comment:
Maybe we don't need the state() function in each class anymore? Or can reduce it?

Because now we can get the information from `data` directly on the `TokenHandler` class level. Same for everything from `feeManager`. The only values local to this class are `underwayFunds` and `totalCredited`.